src/main/java/Duke.java

```diff
@@ -1,10 +1,222 @@
+
+        switch (taskType) {
+            case TODO:
+                if (isDescriptionFilled) {
+                    counter++;
+                    newTask = new ToDo(promptDescription);
+                    tasksList.add(newTask);
+                    systemMessage = "Got it. I've added this task:\n" + "  " + newTask + "\n" + "Now you have "
+                                                + counter + (counter <= 1 ? " task" : " tasks") + " in the list.";
+                    replyFormat(systemMessage);
+                } else {
+                    throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                }
+                break;
+
+            case DEADLINE:
+                if (isDescriptionFilled) {
+                    try {
+                        strings = promptDescription.split(" /by");
+                        description = strings[0];
+                        period = (strings.length > 1 ? strings[1].split(" ",2)[1] : "");
+
+                        if (period != null && !period.trim().isEmpty()) {
+                            newTask = new Deadline(description, period);
+                            tasksList.add(newTask);
+                            counter++;
+                            systemMessage = "Got it. I've added this task:\n" + "  " + newTask + "\n" + "Now you have "
+                                                        + counter + (counter <= 1 ? " task" : " tasks") + " in the list.";
+                            replyFormat(systemMessage);
+                        } else {
+                            throw new DukeException("☹ OOPS!!! The date and/or time cannot be empty.");
+                        }
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("☹ OOPS!!! Please specify the date and/or time in this format:\n"
+                                                            + "  deadline [task description] /by [date and/or time]");
+                    }
+                } else {
+                    throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                }
+                break;
+
+            case EVENT:
+                if (isDescriptionFilled) {
+                    try {
+                        strings = promptDescription.split(" /at");
+                        description = strings[0];
+                        period = (strings.length > 1 ? strings[1].split(" ", 2)[1] : "");
+
+                        if (period != null && !period.trim().isEmpty()) {
+                            newTask = new Event(description, period);
+                            tasksList.add(newTask);
+                            counter++;
+                            systemMessage = "Got it. I've added this task:\n" + "  " + newTask + "\n" + "Now you have "
+                                                        + counter + (counter <= 1 ? " task" : " tasks") + " in the list.";
```

Comment:
The indentation is a bit too excessive here, should only be 8 spaces from the line above